packages/lesmis/src/components/MiniMap.js

```diff
@@ -24,14 +25,38 @@ const Map = styled(MapContainer)`
   height: auto;
 `;
 
+const BasicPolygon = styled(Polygon)`
+  fill: ${RED};
+  fill-opacity: 0.3;
+  stroke: ${RED};
+`;
+
+/* eslint-disable react/prop-types */
+const CountryMask = ({ countryData }) => {
+  return <InversePolygonMask region={countryData?.region} />;
+};
+
+const RegionPolygon = ({ entityData }) => {
+  if (!entityData?.region) return null;
+  if (entityData?.type === 'country') return null; // country is fine without, as it has the mask
+
+  return <BasicPolygon positions={entityData?.region} interactive={false} />;
+};
+
+const PointMarker = ({ entityData }) =>
+  entityData?.point && <IconMarker coordinates={entityData?.point} color={RED} scale={1.5} />;
+/* eslint-enable react/prop-types */
+
 export const MiniMap = ({ entityCode }) => {
+  const { data: countryData, isLoading: isLoadingCountryData } = useEntityData(COUNTRY_CODE);
   const { data: entityData, isLoading: isLoadingEntityData } = useEntityData(entityCode);
 
-  return isLoadingEntityData ? null : (
+  return isLoadingCountryData || isLoadingEntityData ? null : (
```

Comment:
It would be nice if you wanted to, to return a loading state instead of null. Maybe a very light grey box with the same minimum dimensions as the map or a skeleton loader. It would be nice to improve the height change that's happening when the data comes in.

Reply:
Well! Interesting you say that...I played around with this, but it seems all of the nice entity vitals loading skeletons have been broken by some change. It seems that nothing shows up (not even loaders) until useEntityData finishes loading...or something like that. So I pulled my mini-map skeleton loader back out as I couldn't easily tell if it was doing anything. @EMcQ-BES could you look at whether the individual vitals items loading states is still visible?

Admittedly I gave up pretty quickly as it's been hard to fit in dev time today!

Reply:
I would presume that nothing shows up because we use `entity.type` to pick which dashboard/vitals we're going to be displaying

Reply:
Right, that'll be it! So I think that's enough motivation to re-look at changing the useEntityData call to specify whether geo data is required, and if not, initialise from the full entities list. @EMcQ-BES could you make a card?

Reply:
I think it could be solved without that with a bit of shuffling around. So could we make the ticket to polish the loading states and we could try both options. @EMcQ-BES maybe we could pair programme on it?

Reply:
https://github.com/beyondessential/tupaia-backlog/issues/2772

Reply:
@tcaiger Sure we can look into that! I don't have full context for this, I just answered the question I was asked 😅